pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -550,6 +548,16 @@ private void createDefaultValueColumnV1Indices(String column)
         fieldSpec, true/*hasDictionary*/, dictionaryElementSize);
   }
 
+  private boolean isNullable(FieldSpec fieldSpec) {
+    if (_schema.isEnableColumnBasedNullHandling() && fieldSpec.isNullable()) {
+      return true;
+    } else {
+      return _indexLoadingConfig.getTableConfig() != null
+          && _indexLoadingConfig.getTableConfig().getIndexingConfig() != null
+          && _indexLoadingConfig.getTableConfig().getIndexingConfig().isNullHandlingEnabled();
+    }
+  }
```

Comment:
This is incorrect. We want to ignore table based null handling when column based is enabled
```suggestion
    if (_schema.isEnableColumnBasedNullHandling()) {
      return fieldSpec.isNullable();
    } else {
      return _indexLoadingConfig.getTableConfig() != null
          && _indexLoadingConfig.getTableConfig().getIndexingConfig() != null
          && _indexLoadingConfig.getTableConfig().getIndexingConfig().isNullHandlingEnabled();
    }
  }
```